rust/noosphere-cli/src/native/mod.rs

```diff
@@ -29,102 +25,135 @@ use self::{
     },
     workspace::Workspace,
 };
+use anyhow::Result;
+use clap::Parser;
+use noosphere_core::tracing::initialize_tracing;
+use noosphere_storage::StorageConfig;
+
+/// Additional context used to invoke a [Cli] command.
+pub struct CliContext<'a> {
+    /// Path to the current working directory.
+    cwd: PathBuf,
+    /// Path to the global configuration directory, if provided.
+    global_config_dir: Option<&'a Path>,
+}
 
 #[cfg(not(doc))]
 #[allow(missing_docs)]
 pub async fn main() -> Result<()> {
     initialize_tracing(None);
-
-    let workspace = Workspace::new(&std::env::current_dir()?, None)?;
-
-    invoke_cli(Cli::parse(), workspace).await
+    let context = CliContext {
+        cwd: std::env::current_dir()?,
+        global_config_dir: None,
+    };
+    let result = invoke_cli(Cli::parse(), &context).await?;
+    Ok(result)
 }
 
-/// Invoke the CLI implementation imperatively. This is the entrypoint used by
-/// orb when handling a command line invocation. The [Cli] is produced by
-/// parsing the command line arguments, and the [Workspace] is initialized from
-/// the current working directory. The effect of invoking the CLI this way will
-/// depend on the [Cli] and [Workspace] provided (results may vary significantly
-/// depending on those inputs).
-pub async fn invoke_cli(cli: Cli, mut workspace: Workspace) -> Result<()> {
+/// Invoke the CLI implementation imperatively.
+///
+/// This is the entrypoint used by orb when handling a command line invocation.
+/// The [Cli] is produced by parsing the command line arguments, and internally
+/// creates a new [Workspace] from the current working directory.
+///
+/// Use [invoke_cli_with_workspace] if using your own [Workspace].
+pub async fn invoke_cli<'a>(cli: Cli, context: &CliContext<'a>) -> Result<()> {
```

Comment:
`invoke_cli` now creates a `Workspace` from its args rather than accept one so that we can always have the correct storage configuration on instantiation (this is the bulk of the follow up changes)